cbcontainers/remote_configuration/change_applier.go

```diff
@@ -11,6 +11,7 @@ func ApplyConfigChangeToCR(change models.ConfigurationChange, cr *cbcontainersv1
 	if change.AgentVersion != nil {
 		cr.Spec.Version = *change.AgentVersion
 
+		applyAdvancedSettings(change.AdvancedSettings, cr)
```

Comment:
[Nitpick, minor] I'd prefer to have a common pattern and always keep the `cr` as first parameter, makes it easy to follow